packages/react-native/src/theme/createTheme.ts

```diff
@@ -0,0 +1,37 @@
+import deepExtend from 'style-dictionary/lib/utils/deepExtend';
+
+import { defaultTheme } from './defaultTheme';
+import { Theme, StrictTheme } from './types';
+import { setupTokens } from './utils';
+
+/**
+ * This will be used like `const myTheme = createTheme({})`
+ * `myTheme` can then be passed to a Provider
+ * const myTheme = createTheme({})
+ * const myOtherTheme = createTheme({}, myTheme);
+ */
+export const createTheme = (theme?: Theme): StrictTheme => {
+  // merge theme and defaultTheme to get a complete theme
+  // deepExtend is an internal Style Dictionary method
+  // that performs a deep merge on n objects.
+  const mergedTheme: StrictTheme = deepExtend([
+    {},
+    defaultTheme,
+    theme,
+  ]) as StrictTheme;
+  const { colorMode, name, overrides } = mergedTheme;
+
+  // Setting up the tokens.
+  // At the end of this, each token should have a raw value
+  // All references to tokens will be replaced; references to tokens are marked by curly braces
+  const tokens = setupTokens(mergedTheme.tokens);
```

Comment:
Todo: in order to support how overrides work, we will need to conditionally merge override tokens in (if the colorMode is dark and the developer passes in a dark mode override). We don't need to do that for web because of CSS variables.